uvicorn/config.py

```diff
@@ -140,6 +141,11 @@ def __init__(
         else:
             self.reload_dirs = reload_dirs
 
+        if custom_headers is None:
+            self.custom_headers = []
+        else:
+            self.custom_headers = custom_headers
```

Comment:
I'd suggest that in the config, we store the headers as lowercased byte-pairs. It's more efficient for us to do that conversion here, just the once.

I guess we want something like this?...

```python
if headers is None:
    headers = []
self.headers = [
    (key.lower().encode('latin-1'), value.encode('latin-1'))
    for key, value in headers
]
if b'server' not in dict(self.headers):
    self.headers = [(b'server', b'uvicorn')] + self.headers
```

Reply:
Don't worry about `date` - it doesn't make sense for users to specify that, and if they want to do so. There's plenty of header values that it'd be erronous for users to setup here - if they're setting custom headers it's up to them to do so sensibly.

Reply:
I'm always reticent to store much business logic in `__init__`. The stuff I put in there already was because I couldn't think of another way to handle it. But, if you'd like it that way, I can accommodate.

Reply:
I found a compromise. I'm going to perform the logic in `config.load()`.